threads.js

```diff
@@ -2138,7 +2138,8 @@ Process.prototype.reportURL = function (url) {
     var response;
     if (!this.httpRequest) {
         this.httpRequest = new XMLHttpRequest();
-        this.httpRequest.open("GET", 'http://' + url, true);
+        var protocol = document.location.protocol || 'http:';
```

Comment:
We would probably want to check if this is `file:` and convert that to http. I do think people make http requests from regular file pages. It's not technically the most supported mode of using Snap<em>!</em> but it does happen.